### PR TITLE
fix saved values for on("binding::x") event callback

### PR DESCRIPTION
**Pull request by Agyar**
current code was trying to access missing value in dict ("mods"), and
stored values where not sufficient to actually act on specific bindings

added example to change transparency of focused window using bindings

### CODE REVIEW

File: i3ipc/i3ipc.py [hunk 1]
```diff
@@ -253,9 +253,9 @@ def __init__(self, data):
 class BindingInfo(object):
     def __init__(self, data):
         self.command = data['command']
-        self.mods = data['mods']
+        self.mods = data['event_state_mask']
         self.input_code = data['input_code']
-        self.symbol = data['symbol']
+        self.symbols = data['symbols']
```
Comment (acrisci) [inline i3ipc/i3ipc.py]: when did this change?

**Comment by acrisci** [comment]
This is a breaking change. Can you keep backwards compatibility somehow?

**Comment by Agyar** [comment]
I can for some part, but the part where we have 
```
Self.mods = data['mods']
```
Was failing and this is the very reason of this pr. But maybe I missed something?

**Comment by Agyar** [comment]
For the symbol, symbols, we could have the two ? I don't remember if I checked that the previous code was correctly returning all expected values.

**Comment by acrisci** [comment]
We need some tests for this we can run on different versions of i3.

**Comment by Agyar** [comment]
I'll see what I can do next week probably.
(Sway user here)

**Comment by acrisci** [comment]
i3 is still using `symbol` not `symbols`. Open an issue on sway to fix compatibility with i3.

**Comment by lasers** [comment]
Bump.

**Comment by acrisci** [comment]
@lasers 

> Open an issue on sway to fix compatibility with i3.

**Comment by lasers** [comment]
@acrisci 

- [x] Open an issue on sway to fix compatibility with i3.

**Comment by acrisci** [comment]
I'm looking at the py3status issue now, and the issue is not `symbols`, it is that `mods` has been replaced by `event_state_mask`. That's worth a sway issue, but I feel ok defaulting a value that's not provided.

However I will not combine values for compatibility.

**Comment by lasers** [comment]
I asked about ~~mods~~ event_state_mask. See the issue again. Maybe you can start using that?

**Comment by acrisci** [comment]
Yeah, this PR needs to be updated then to maintain backwards compatibility.

**Comment by lasers** [comment]
My dear @Agyar, can you update this PR when you are able to? Pretty please... and thank you.

**Comment by Agyar** [comment]
I would with great pleasure but I didn't get much time to give this PR the love it deserves so sadly I can't make any promises (working + finishing a Phd and being a young dad doesn't help much)

**Comment by acrisci** [comment]
No problem, thanks for bringing this to my attention.

**Comment by acrisci** [comment]
262246d

**Pull request closed without merging**